dynamiqs/mesolve/rouchon.py

```diff
@@ -2,20 +2,24 @@
 
 from math import sqrt
 
+import torch
 from torch import Tensor
+from torch.linalg import cholesky_ex as cholesky
 
 from ..solvers.ode.fixed_solver import AdjointFixedSolver
-from ..solvers.utils import (
-    CallableTDTensor,
-    ConstantTDTensor,
-    cache,
-    inv_sqrtm,
-    kraus_map,
-)
+from ..solvers.utils import cache, inv_sqrtm, kraus_map
 from ..utils.utils import unit
 from .me_solver import MESolver
 
 
+def inv_mm_invmh(A: Tensor, B: Tensor) -> Tensor:
```

Comment:
Probably a simple fix is to add the option `upper=True` in this function.

Reply:
How about `inv_kraus_matmul` for the name of this function? Not sure if it's better. I'll let you decide.